open-api/rest-catalog-open-api.yaml

```diff
@@ -120,6 +122,32 @@ paths:
         5XX:
           $ref: '#/components/responses/ServerErrorResponse'
 
+  /v1/oauth/token:
+
+    post:
+      tags:
+        - Auth API
+      summary: Get a token using an OAuth2 flow
+      operationId: getToken
+      description:
+        Exchange credentials for a token using the OAuth2 client credentials flow or token exchange.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TokenRequest'
+      responses:
+        200:
+          $ref: '#/components/responses/TokenResponse'
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        401:
```

Comment:
The OAuth2 RFC also specifies what the error response body should look like. We'd need to add an `error` field containing one of the codes if we want to be strict about that.

Reply:
(Here's the relevant [section](https://datatracker.ietf.org/doc/html/rfc6749#section-5.2) in the RFC)

Reply:
I added an OAuthErrorResponse to use instead of the standard one.